runtime/common/logging_xwalk.cc

```diff
@@ -0,0 +1,205 @@
+// InitXwalkLogging() and the beginning of CleanupXwalkLogging().
+bool xwalk_logging_initialized_ = false;
+
+// Set if we called InitXwalkLogging() but failed to initialize.
+bool xwalk_logging_failed_ = false;
+
+// This should be true for exactly the period between the end of
+// InitXwalkLogging() and the beginning of CleanupXwalkLogging().
+bool xwalk_logging_redirected_ = false;
+
+#if defined(OS_WIN)
+// {7FE69228-633E-4f06-80C1-527FEA23E3A7}
+const GUID kXwalkTraceProviderName = {
+    0x7fe69228, 0x633e, 0x4f06,
+        { 0x80, 0xc1, 0x52, 0x7f, 0xea, 0x23, 0xe3, 0xa7 } };
+#endif
+
+}  // anonymous namespace
+
+namespace logging {
+
+LoggingDestination DetermineLogMode(const base::CommandLine& command_line) {
+  // only use OutputDebugString in debug mode
+#ifdef NDEBUG
+  bool enable_logging = false;
+  const char *kInvertLoggingSwitch = switches::kEnableLogging;
+  const logging::LoggingDestination kDefaultLoggingMode = logging::LOG_TO_FILE;
+#else
+  bool enable_logging = true;
+  const char *kInvertLoggingSwitch = switches::kDisableLogging;
+  const logging::LoggingDestination kDefaultLoggingMode = logging::LOG_TO_ALL;
+#endif
+
+  if (command_line.HasSwitch(kInvertLoggingSwitch))
+    enable_logging = !enable_logging;
+
+  logging::LoggingDestination log_mode;
+  if (enable_logging) {
+    // Let --enable-logging=stderr force only stderr, particularly useful for
+    // non-debug builds where otherwise you can't get logs to stderr at all.
+    if (command_line.GetSwitchValueASCII(switches::kEnableLogging) == "stderr")
+      log_mode = logging::LOG_TO_SYSTEM_DEBUG_LOG;
+    else
+      log_mode = kDefaultLoggingMode;
+  } else {
+    log_mode = logging::LOG_NONE;
+  }
+  return log_mode;
+}
+
+void InitXwalkLogging(const base::CommandLine& command_line,
+                      OldFileDeletionState delete_old_log_file) {
+  DCHECK(!xwalk_logging_initialized_) <<
+      "Attempted to initialize logging when it was already initialized.";
+
+  LoggingDestination logging_dest = DetermineLogMode(command_line);
+  LogLockingState log_locking_state = LOCK_LOG_FILE;
+  base::FilePath log_path;
+
+  // Don't resolve the log path unless we need to. Otherwise we leave an open
+  // ALPC handle after sandbox lockdown on Windows.
+  if ((logging_dest & LOG_TO_FILE) != 0) {
+    log_path = GetLogFileName();
+  } else {
+    log_locking_state = DONT_LOCK_LOG_FILE;
+  }
+
+  logging::LoggingSettings settings;
+  settings.logging_dest = logging_dest;
+  settings.log_file = log_path.value().c_str();
+  settings.lock_log = log_locking_state;
+  settings.delete_old = delete_old_log_file;
+  bool success = logging::InitLogging(settings);
+
+  if (!success) {
+    DPLOG(ERROR) << "Unable to initialize logging to " << log_path.value();
+    xwalk_logging_failed_ = true;
+    return;
+  }
+
+  // Default to showing error dialogs.
+  if (!base::CommandLine::ForCurrentProcess()->HasSwitch(
+      switches::kNoErrorDialogs))
```

Comment:
I tried to build on Linux with <code>component=shared_library</code>. But it built successfully and didn't meet linking error. Here is my result:
```
jj@jj-desktop:/media/jiajia/p1/project/crosswalk-1/src/xwalk$ python gyp_xwalk.py -Dcomponent=shared_library -Dmediacodecs_EULA=1
Updating projects from gyp files...
jj@jj-desktop:/media/jiajia/p1/project/crosswalk-1/src/xwalk$ ninja -C ../out/Debug xwalk
ninja: Entering directory `../out/Debug'
[13287/13287] LINK xwalk
jj@jj-desktop:/media/jiajia/p1/project/crosswalk-1/src/xwalk$ echo $?
0
```
What's the steps to reproduce what you said?